spec/test_prof/cops/rspec/aggregate_examples_spec.rb

```diff
@@ -0,0 +1,620 @@
+# frozen_string_literal: true
+
+require "cop_helper"
+require "test_prof/cops/rspec/aggregate_examples"
+
+RSpec.describe RuboCop::Cop::RSpec::AggregateExamples, :config do
+  subject(:cop) { described_class.new(config) }
+
+  shared_examples "detects and autocorrects in example group" do |group|
+    context "with '#{group}'" do
+      offensive_source = <<-RUBY
+        #{group} 'some docstring' do
+          it { is_expected.to be_awesome }
+          it { expect(subject).to be_amazing }
+          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Aggregate with the example at line 2.
+          it { expect(article).to be_brilliant }
+          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Aggregate with the example at line 2.
+        end
+      RUBY
+
+      good_source = <<-RUBY
+        #{group} 'some docstring' do
+          specify do
+            is_expected.to be_awesome
+            expect(subject).to be_amazing
+            expect(article).to be_brilliant
+          end
+        end
+      RUBY
+
+      it "detects and autocorrects" do
+        expect_offense(offensive_source)
+        expect_correction(good_source)
+      end
+    end
+  end
+
+  # Detects aggregatable examples inside all aliases of example groups.
+  it_behaves_like "detects and autocorrects in example group", :context
+  it_behaves_like "detects and autocorrects in example group", :describe
+  it_behaves_like "detects and autocorrects in example group", :feature
+  it_behaves_like "detects and autocorrects in example group",
+    :example_group
+
+  # Non-expectation statements can have side effects, when e.g. being
+  # part of the setup of the example.
+  # Examples containing expectations wrapped in a method call, e.g.
+  # `expect_no_corrections` are not considered aggregatable.
+  context "with examples with non-expectation statements" do
+    it "does not detect offenses" do
+      expect_no_offenses(<<-RUBY)
```

Comment:
Non-squiggly HEREDOC produces output that is hard to read due to excessive and inconsistent identation.